File: MatrixMult/MatrizMult.java
import java.util.Arrays;

public class MatrizMult{
    public static int[][] multiplicarMatrices(int[][] A, int[][] B){
        // Verifica si el número de columnas de la matriz A es igual al número de filas de la matriz B
        if (A[0].length != B.length){
            throw new IllegalArgumentException("Las dimensiones de las matrices no son compatibles para la multiplicación.");
        }

        // Dimensiones de la matriz resultante
        int filas_A = A.length;
        int columnas_A = A[0].length;
        int columnas_B = B[0].length;

        // Crear la matriz resultante inicializada en 0
        int[][] resultado = new int[filas_A][columnas_B];

        // Multiplicación de matrices
        for (int i = 0; i < filas_A; i++){
            for (int j = 0; j < columnas_B; j++){
                for (int k = 0; k < columnas_A; k++){
                    resultado[i][j] += A[i][k] * B[k][j];
                }
            }
        }

        return resultado;
    }

    public static void main(String[] args){
        // Definir dos matrices
        int[][] A = {
            {1, 2, 3},
            {4, 5, 6}
        };

        int[][] B = {
            {7, 8},
            {9, 10},
            {11, 12}
        };

        // Multiplicar las matrices
        try{
            int[][] resultado = multiplicarMatrices(A, B);
            System.out.println("Resultado de la multiplicación de matrices:");
            for (int[] fila : resultado){
                System.out.println(Arrays.toString(fila));
            }
        }catch (IllegalArgumentException e){
            System.out.println(e.getMessage());
        }
    }
}

File: MatrixMult/matrizMult.py

def multiplicar_matrices(A, B):
    # Verifica si el número de columnas de la matriz A es igual al número de filas de la matriz B
    if len(A[0]) != len(B):
        raise ValueError("Las dimensiones de las matrices no son compatibles para la multiplicación.")

    # Dimensiones de la matriz resultante
    filas_A = len(A)
    columnas_A = len(A[0])
    columnas_B = len(B[0])

    # Crear la matriz resultante inicializada en 0
    resultado = [[0 for _ in range(columnas_B)] for _ in range(filas_A)]

    # Multiplicación de matrices
    for i in range(filas_A):
        for j in range(columnas_B):
            for k in range(columnas_A):
                resultado[i][j] += A[i][k] * B[k][j]

    return resultado


# Definir dos matrices
A = [
    [1, 2, 3],
    [4, 5, 6]
]

B = [
    [7, 8],
    [9, 10],
    [11, 12]
]

# Multiplicar las matrices
try:
    resultado = multiplicar_matrices(A, B)
    print("Resultado de la multiplicación de matrices:")
    for fila in resultado:
        print(fila)
except ValueError as e:
    print(e)
